### PR TITLE
Migrate default plugins to TypeScript

### PR DESCRIPTION
**Pull request by jovyntls**
**What is the purpose of this pull request?**

- [ ] Documentation update
- [ ] Bug fix
- [ ] Feature addition or enhancement
- [x] Code maintenance
- [ ] DevOps
- [ ] Improve developer experience
- [ ] Others, please explain:

Part of #1913

**Overview of changes:**
Migrates 4 default plugins to TS: 
- `markbind-plugin-anchors.js`
- `markbind-plugin-plantuml.js`
- `markbind-plugin-shorthandSyntax.js`
- `markbind-plugin-tree.js`

**Anything you'd like to highlight/discuss:**
N/A

**Testing instructions:**
N/A

**Proposed commit message: (wrap lines at 72 characters)**
(rebase commit)

---

**Checklist:** :ballot_box_with_check:

- [ ] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [x] Linked all related issues
- [x] No unrelated changes

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -76,6 +76,7 @@
   "devDependencies": {
     "@types/bluebird": "^3.5.36",
     "@types/cheerio": "^0.22.31",
+    "@types/crypto-js": "^4.1.1",
```
[the remark this answers is not in the document]
Comment (jovyntls) [reply packages/core/package.json:79]: cypto-js was used in one of the plugins ([here](https://github.com/jovyntls/markbind/blob/43071db91025f4f5a3589c4366d2d632e5801574/packages/core/src/plugins/default/markbind-plugin-plantuml.ts#L10)) - without importing its types the linter complains that it implicitly has an 'any' type

File: packages/core/src/plugins/default/markbind-plugin-plantuml.ts [hunk 2]
```diff
@@ -89,12 +92,12 @@ module.exports = {
     graphvizCheckCompleted = false;
   },
 
-  processNode: (pluginContext, node, config) => {
-    if (node.name !== 'puml') {
+  processNode: (_pluginContext: PluginContext, node: MbNode, config: NodeProcessorConfig) => {
+    if (!node.name || node.name !== 'puml') {
```
Comment (raysonkoh) [inline packages/core/src/plugins/default/markbind-plugin-plantuml.ts:96]: Is this check needed since `node.name` will not be undefined in `MbNode`?

Comment (jovyntls) [reply packages/core/src/plugins/default/markbind-plugin-plantuml.ts:96]: Thanks for the catch! Removed the check

**Review by raysonkoh — APPROVED** [review]
LGTM

**Review by tlylt — COMMENTED** [review]
Thanks @jovyntls, I think looks great!

**Pull request merged**